### PR TITLE
add telemetry events for ppr & staleTimes experimental flags

### PR DESCRIPTION
**Pull request by ztanner**
Adds `NextConfig.experimental.ppr` and `NextConfig.experimental.staletimes` to the `NEXT_BUILD_FEATURE_USAGE` telemetry event. 

Closes NEXT-2981

### CODE REVIEW

**Comment by ztanner** [comment]
* **#63981** <a href="https://app.graphite.dev/github/pr/vercel/next.js/63981?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @ztanner and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by ijjk** [comment]
## Failing test suites 
Commit: fdc040fa74eebf14ce30e49617e3025fa7a1f128

`pnpm test test/integration/create-next-app/templates-app.test.ts`
- create-next-app --app > should create JS appDir projects with --js --src-dir

<details>
<summary>Expand output</summary>

  ● create-next-app --app › should create JS appDir projects with --js --src-dir

    thrown: "Exceeded timeout of 60000 ms for a test.
    Add a timeout value to this test to increase the timeout, if this is a long-running test. See https://jestjs.io/docs/api#testname-fn-timeout."

      123 |   })
      124 |
    > 125 |   it('should create JS appDir projects with --js --src-dir', async () => {
          |   ^
      126 |     await useTempDir(async (cwd) => {
      127 |       const projectName = 'appdir-test'
      128 |       const childProcess = createNextApp(

      at it (integration/create-next-app/templates-app.test.ts:125:3)
      at Object.describe (integration/create-next-app/templates-app.test.ts:52:1)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

**Comment by ijjk** [comment]
# Stats from current PR

<details open>
<summary><strong>Default Build</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | vercel/next.js 04-02-add_telemetry_events_for_ppr_staleTimes_experimental_flags | Change |
| - | - | - | - |
| buildDuration | 13.8s | 13.9s | N/A |
| buildDurationCached | 7.4s | 6.1s | N/A |
| nodeModulesSize | 199 MB | 199 MB | ⚠️ +2.98 kB |
| nextStartRea..uration (ms) | 401ms | 399ms | N/A |

</details>

<details>
<summary><strong>Client Bundles (main, webpack)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 04-02-add_telemetry_events_for_ppr_staleTimes_experimental_flags | Change |
| - | - | - | - |
| 2453-HASH.js gzip | 31.4 kB | 31.4 kB | N/A |
| 3304.HASH.js gzip | 181 B | 181 B | ✓ |
| 3f784ff6-HASH.js gzip | 53.7 kB | 53.7 kB | ✓ |
| 8299-HASH.js gzip | 5.04 kB | 5.04 kB | N/A |
| framework-HASH.js gzip | 45.2 kB | 45.2 kB | ✓ |
| main-app-HASH.js gzip | 244 B | 241 B | N/A |
| main-HASH.js gzip | 32.2 kB | 32.2 kB | N/A |
| webpack-HASH.js gzip | 1.68 kB | 1.68 kB | N/A |
| Overall change | 99 kB | 99 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 04-02-add_telemetry_events_for_ppr_staleTimes_experimental_flags | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 04-02-add_telemetry_events_for_ppr_staleTimes_experimental_flags | Change |
| - | - | - | - |
| _app-HASH.js gzip | 196 B | 197 B | N/A |
| _error-HASH.js gzip | 184 B | 184 B | ✓ |
| amp-HASH.js gzip | 505 B | 505 B | ✓ |
| css-HASH.js gzip | 324 B | 325 B | N/A |
| dynamic-HASH.js gzip | 2.5 kB | 2.5 kB | N/A |
| edge-ssr-HASH.js gzip | 258 B | 258 B | ✓ |
| head-HASH.js gzip | 352 B | 352 B | ✓ |
| hooks-HASH.js gzip | 370 B | 371 B | N/A |
| image-HASH.js gzip | 4.21 kB | 4.21 kB | ✓ |
| index-HASH.js gzip | 259 B | 259 B | ✓ |
| link-HASH.js gzip | 2.67 kB | 2.67 kB | N/A |
| routerDirect..HASH.js gzip | 314 B | 312 B | N/A |
| script-HASH.js gzip | 386 B | 386 B | ✓ |
| withRouter-HASH.js gzip | 309 B | 309 B | ✓ |
| 1afbb74e6ecf..834.css gzip | 106 B | 106 B | ✓ |
| Overall change | 6.57 kB | 6.57 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 04-02-add_telemetry_events_for_ppr_staleTimes_experimental_flags | Change |
| - | - | - | - |
| _buildManifest.js gzip | 481 B | 484 B | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 04-02-add_telemetry_events_for_ppr_staleTimes_experimental_flags | Change |
| - | - | - | - |
| index.html gzip | 528 B | 529 B | N/A |
| link.html gzip | 540 B | 541 B | N/A |
| withRouter.html gzip | 523 B | 523 B | ✓ |
| Overall change | 523 B | 523 B | ✓ |

</details>

<details>
<summary><strong>Edge SSR bundle Size</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 04-02-add_telemetry_events_for_ppr_staleTimes_experimental_flags | Change |
| - | - | - | - |
| edge-ssr.js gzip | 95.4 kB | 95.4 kB | N/A |
| page.js gzip | 3.06 kB | 3.06 kB | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Middleware size</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 04-02-add_telemetry_events_for_ppr_staleTimes_experimental_flags | Change |
| - | - | - | - |
| middleware-b..fest.js gzip | 624 B | 624 B | ✓ |
| middleware-r..fest.js gzip | 151 B | 151 B | ✓ |
| middleware.js gzip | 25.5 kB | 25.5 kB | N/A |
| edge-runtime..pack.js gzip | 839 B | 839 B | ✓ |
| Overall change | 1.61 kB | 1.61 kB | ✓ |

</details>

<details>
<summary><strong>Next Runtimes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 04-02-add_telemetry_events_for_ppr_staleTimes_experimental_flags | Change |
| - | - | - | - |
| app-page-exp...dev.js gzip | 170 kB | 170 kB | ✓ |
| app-page-exp..prod.js gzip | 97 kB | 97 kB | ✓ |
| app-page-tur..prod.js gzip | 98.8 kB | 98.8 kB | ✓ |
| app-page-tur..prod.js gzip | 93 kB | 93 kB | ✓ |
| app-page.run...dev.js gzip | 144 kB | 144 kB | ✓ |
| app-page.run..prod.js gzip | 91.5 kB | 91.5 kB | ✓ |
| app-route-ex...dev.js gzip | 21.4 kB | 21.4 kB | ✓ |
| app-route-ex..prod.js gzip | 15.2 kB | 15.2 kB | ✓ |
| app-route-tu..prod.js gzip | 15.2 kB | 15.2 kB | ✓ |
| app-route-tu..prod.js gzip | 14.9 kB | 14.9 kB | ✓ |
| app-route.ru...dev.js gzip | 21.1 kB | 21.1 kB | ✓ |
| app-route.ru..prod.js gzip | 14.9 kB | 14.9 kB | ✓ |
| pages-api-tu..prod.js gzip | 9.55 kB | 9.55 kB | ✓ |
| pages-api.ru...dev.js gzip | 9.82 kB | 9.82 kB | ✓ |
| pages-api.ru..prod.js gzip | 9.55 kB | 9.55 kB | ✓ |
| pages-turbo...prod.js gzip | 22.5 kB | 22.5 kB | ✓ |
| pages.runtim...dev.js gzip | 23.1 kB | 23.1 kB | ✓ |
| pages.runtim..prod.js gzip | 22.4 kB | 22.4 kB | ✓ |
| server.runti..prod.js gzip | 51 kB | 51 kB | ✓ |
| Overall change | 945 kB | 945 kB | ✓ |

</details>

<details>
<summary><strong>build cache</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | vercel/next.js 04-02-add_telemetry_events_for_ppr_staleTimes_experimental_flags | Change |
| - | - | - | - |
| 0.pack gzip | 1.58 MB | 1.58 MB | ⚠️ +537 B |
| index.pack gzip | 107 kB | 106 kB | N/A |
| Overall change | 1.58 MB | 1.58 MB | ⚠️ +537 B |

</details>

<details>
<summary><strong>Diff details</strong></summary>

<details>
<summary>Diff for <strong>middleware.js</strong></summary>

Diff too large to display
</details>

</details>

</details>
Commit: fdc040fa74eebf14ce30e49617e3025fa7a1f128

File: packages/next/src/build/index.ts [hunk 1]
```diff
@@ -2421,6 +2421,18 @@ export default async function build(
...
         },
+        {
+          featureName: 'experimental/ppr',
+          invocationCount: config.experimental.ppr ? 1 : 0,
+        },
+        {
+          featureName: 'experimental/staleTimes',
+          invocationCount:
+            config.experimental.staleTimes !==
+            defaultConfig.experimental?.staleTimes
```
Comment (ijjk) [inline packages/next/src/build/index.ts:2432]: Any reason we don't want the actual stale time values?

**Review by ijjk — APPROVED** [review]

**Pull request merged**